packages/fx-core/src/plugins/resource/apiconnector/plugin.ts

```diff
@@ -399,23 +401,14 @@ export class ApiConnectorImpl {
         );
       }
     }
-    const whichComponent = new QTreeNode({
-      name: Constants.questionKey.componentsSelect,
-      type: "multiSelect",
-      staticOptions: componentOptions,
-      title: getLocalizedString("plugins.apiConnector.whichService.title"),
-      validation: {
-        validFunc: checkEmptySelect,
-      },
-      placeholder: getLocalizedString("plugins.apiConnector.whichService.placeholder"), // Use the placeholder to display some description
-    });
+    const whichComponent = new ComponentsQuestion(ctx, componentOptions, projectPath as string);
```

Comment:
is it possible to get componentOptions inside `ComponentsQuestion` to make this function cleaner?

Reply:
updated